pkg/services/serviceaccounts/database/database_test.go

```diff
@@ -110,6 +111,69 @@ func TestStore_RetrieveServiceAccount(t *testing.T) {
 	}
 }
 
+func TestStore_AddServiceAccountToTeam(t *testing.T) {
+	cases := []struct {
+		desc                        string
+		sa                          tests.TestUser
+		team                        tests.TestTeam
+		expectedNumberOfTeamMembers int
+		expectedErr                 error
+	}{
+		{
+			desc: "should be able to add service account to team",
+			sa: tests.TestUser{
+				Login:            "servicetest1@admin",
+				IsServiceAccount: true,
+				Permissions: []accesscontrol.Permission{
+					// {
+					// 	Action: accesscontrol.ActionTeamsPermissionsWrite,
+					// 	Scope:  accesscontrol.ScopeTeamsAll,
+					// },
+					{
+						Action: accesscontrol.ActionOrgUsersRead,
+						Scope:  accesscontrol.ScopeUsersAll,
+					},
+				},
+			},
+			team:                        tests.TestTeam{Email: "teamtest1@admin", Name: "test-team"},
+			expectedNumberOfTeamMembers: 1,
+			expectedErr:                 nil,
+		},
+	}
+	for _, c := range cases {
+		t.Run(c.desc, func(t *testing.T) {
+			db, store := setupTestDatabase(t)
+			sa := tests.SetupUserServiceAccount(t, db, c.sa)
+			team, err := db.CreateTeam(c.team.Name, c.team.Email, sa.OrgId)
+			if err != nil {
+				require.NoError(t, err)
+			}
+			cmd := &serviceaccounts.AddServiceAccountToTeamCommand{OrgId: sa.OrgId, TeamId: team.Id}
+			err = store.AddServiceAccountToTeam(context.Background(), sa.Id, cmd)
+			if c.expectedErr != nil {
+				require.ErrorIs(t, err, c.expectedErr)
+			}
+
+			signedInUser := &models.SignedInUser{
+				UserId: sa.Id,
+				OrgId:  sa.OrgId,
+				Permissions: map[int64]map[string][]string{
+					sa.OrgId: {
+						accesscontrol.ActionTeamsRead:    []string{accesscontrol.ScopeTeamsAll},
+						accesscontrol.ActionOrgUsersRead: []string{accesscontrol.ScopeUsersAll},
+					},
+				},
+			}
+			teamQuery := models.GetTeamMembersQuery{OrgId: sa.OrgId, TeamId: team.Id, SignedInUser: signedInUser}
+			err = db.GetTeamMembers(context.Background(), &teamQuery)
+			require.NoError(t, err)
+			require.Equal(t, c.expectedNumberOfTeamMembers, len(teamQuery.Result))
```

Comment:
@Jguer running the tests yields

```bash
go test -count 1 -v ./pkg/services/serviceaccounts/database -run TestStore_AddServiceAccountToTeam
```

=== RUN   TestStore_AddServiceAccountToTeam/should_be_able_to_add_service_account_to_team
    database_test.go:170:
        	Error Trace:	database_test.go:170
        	Error:      	Not equal:
        	            	expected: 1
        	            	actual  : 0